### PR TITLE
[SYCL] Disable deferred buffer release for interop handle usage

### PR DESCRIPTION
**Pull request by KseniyaTikhomirova**
non-blocking buffer destructor could not be used with L0 interop handles with ownership "keep" since no way to distinguish when all resources is released by RT.

### CODE REVIEW

**Comment by KseniyaTikhomirova** [comment]
********************
Failed Tests (1):
  SYCL :: Graph/submission_while_executing.cpp

is unrelated, reported here https://github.com/intel/llvm/issues/11277

**Comment by KseniyaTikhomirova** [comment]
Failed Tests (1):
  SYCL :: Graph/submission_while_executing.cpp


Testing Time: 2738.61s
  Unsupported      : 761
  Passed           : 918
  Expectedly Failed:  26
  Failed           :   1

**Comment by KseniyaTikhomirova** [comment]
failure is already reported here https://github.com/intel/llvm/issues/11277

**Comment by KseniyaTikhomirova** [comment]
@cperkinsintel kindly ping

File: sycl/source/detail/context_impl.hpp [hunk 1]
```diff
@@ -241,9 +243,12 @@ class context_impl {
                        const std::set<std::uintptr_t> &ImgIdentifiers,
                        const std::string &ObjectTypeName);
 
+  bool isOwnedByRuntime() { return MOwnedByRuntime; };
+
   enum PropertySupport { NotSupported = 0, Supported = 1, NotChecked = 2 };
 
 private:
+  bool MOwnedByRuntime;
```
Comment (cperkinsintel) [inline sycl/source/detail/context_impl.hpp:251]: I like this name much better than "KeepOwnership" which was always confusing to me.  Kept by whom?    

Thanks!

**Comment by KseniyaTikhomirova** [comment]
> Since this is a test involving L0 ownership, shouldn't it have at least one E2E test?

there is no way to check if destructor was blocking or not in E2E test. I could check this mechanism only via unittests.

**Comment by KseniyaTikhomirova** [comment]
@cperkinsintel are you ok to merge it?

**Comment by KseniyaTikhomirova** [comment]
@cperkinsintel kindly ping

**Comment by cperkinsintel** [comment]
I'm Ok to merge it

**Pull request merged**